### PR TITLE
Correctly handle parsing of JSON strings containing single quotes.

### LINKED ISSUE #21147: JsonObjectParser mishandles single quotes in 2.4.9-RC2
The `JsonObjectParser` used internally by the `JsonEntityStreamingSupport` framing decoder fails when object keys or values contain unmatched single quotes, e.g:

```json
{
    "text": "'"
}
```

The most obvious culprit appears to be [here](https://github.com/akka/akka/blob/a712f0149a38335c77dbed9d33da22a7a083c8eb/akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala#L20):

    final val DoubleQuote = '\''.toByte

In the case of a single object array such as `[{"text": "'"}]`, the framing decoder will simply not emit the object.

### PR DESCRIPTION
**Pull request by mikesname**
Fixes #21147.

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala [hunk 1]
```diff
@@ -17,7 +17,7 @@ private[akka] object JsonObjectParser {
   final val SquareBraceEnd = ']'.toByte
   final val CurlyBraceStart = '{'.toByte
   final val CurlyBraceEnd = '}'.toByte
-  final val DoubleQuote = '\''.toByte
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala]: oh wow :)

**Comment by drewhk** [comment]
LGTM

**Comment by johanandren** [comment]
LGTM, thanks!

**Comment by johanandren** [comment]
OK TO TEST

File: akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala (hunk 1 above)
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala]: Whoa, thanks for noticing! LGTM

**Comment by patriknw** [comment]
LGTM

**Pull request merged**